keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,1319 @@
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+#### Alpha Goals
+
+1.	Provide Initial infrastructure to support CCI Driver Plugins and k8s Deployments requiring CCI Drivers after scheduling
+2.  Provide a feature gate to enable CCI Manager. The feature will require CPU Manager policy set to None.
+3.  Provide a CCI test driver which can demonstrate CCI Driver implementation requirementes and several resource management use-cases for cpu and memory.
+4.  Provide documntation of the CCI Manager and the provided test driver plus illustration of the currenlty covered use-cases.
+5.	Support seamless k8s start.
+6.	CCI Driver results (cpusets and memory affinity) are passed back to kubelet and allocated.
+7.	Handle resource management state for cpu and memory through CCI. 
+8.	Support proper fail mechanisms of Pods if CCI driver is not available - Error message + retry 
+9.	Continue cluster operation if CCI driver is not available
+10.  Podspecs:  Will be able to support current pod specs.  While there may be additional 
+extensibility in the future, the current pod specs will still work.
+
+
+#### Beta & Post-Beta Goals
+
+1.	Support standard pod deployments not requiring CCI Drivers through CPU & Memory Manager: all QoS types
+2.  Support hint providers for topology manager
+3.  Interoperability with Device Plugins, DRA, et cetera. 
+4.	Identify and provide a minimal in-cluster operational core (from existing CPU Manager, Memory Manager, Topology Manager) either through in-tree libraries or through further KEPs. 
+5.  Replace CCI policy with a component integrated in None and Static CPU Manager policies.
+6.  Gather feedback on the feature through surveys.
+7.	E2E testing including other components.
+8.  Adding support for e2e tests in testgrid.
+9.  Health identification mechanisms and automatic health-repair procedures for CCI Drivers.  
+8.  Optimize on-node resource usage (approaches to avoid waste of resources).
+
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* Break any existing use cases for topology manager, memory manager, cpu manager, or 
+device manager.
+* Creating any more latency than there is today for scheduling:  We should be 
+careful not to add any latency into scheduling over what exists today for default behavior.
+* CPU & Memory Management on scheduler side.
+* Optimal binpacking of resources (we will still consider if we can optimize resource utilization in Beta).
+
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
```

Comment:
We should add a very high level proposal here.

Reply:
Added.